### PR TITLE
Make horses and llama's fire LivingFallEvent.

### LINKED ISSUE #13: Suggestion about LivingFallEvent
Original Forge doesn't call LivingFallEvent when a horse(or a llama) falls. Could this be fixed in Cleanroom?

See:
AbstractHorse#fall
```java
public void fall(float distance, float damageMultiplier)
    {
        if (distance > 1.0F)
        {
            this.playSound(SoundEvents.ENTITY_HORSE_LAND, 0.4F, 1.0F);
        }

        int i = MathHelper.ceil((distance * 0.5F - 3.0F) * damageMultiplier);

        if (i > 0)
        {
            this.attackEntityFrom(DamageSource.FALL, (float)i);

            if (this.isBeingRidden())
            {
                for (Entity entity : this.getRecursivePassengers())
                {
                    entity.attackEntityFrom(DamageSource.FALL, (float)i);
                }
            }

            IBlockState iblockstate = this.world.getBlockState(new BlockPos(this.posX, this.posY - 0.2D - (double)this.prevRotationYaw, this.posZ));
            Block block = iblockstate.getBlock();

            if (iblockstate.getMaterial() != Material.AIR && !this.isSilent())
            {
                SoundType soundtype = block.getSoundType();
                this.world.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, soundtype.getStepSound(), this.getSoundCategory(), soundtype.getVolume() * 0.5F, soundtype.getPitch() * 0.75F);
            }
        }
    }
```

EntityLivingBase#fall
```java
public void fall(float distance, float damageMultiplier)
    {
        float[] ret = net.minecraftforge.common.ForgeHooks.onLivingFall(this, distance, damageMultiplier);
        if (ret == null) return;
        distance = ret[0]; damageMultiplier = ret[1];
        super.fall(distance, damageMultiplier);
        PotionEffect potioneffect = this.getActivePotionEffect(MobEffects.JUMP_BOOST);
        float f = potioneffect == null ? 0.0F : (float)(potioneffect.getAmplifier() + 1);
        int i = MathHelper.ceil((distance - 3.0F - f) * damageMultiplier);

        if (i > 0)
        {
            this.playSound(this.getFallSound(i), 1.0F, 1.0F);
            this.attackEntityFrom(DamageSource.FALL, (float)i);
            int j = MathHelper.floor(this.posX);
            int k = MathHelper.floor(this.posY - 0.20000000298023224D);
            int l = MathHelper.floor(this.posZ);
            IBlockState iblockstate = this.world.getBlockState(new BlockPos(j, k, l));

            if (iblockstate.getMaterial() != Material.AIR)
            {
                SoundType soundtype = iblockstate.getBlock().getSoundType(iblockstate, world, new BlockPos(j, k, l), this);
                this.playSound(soundtype.getFallSound(), soundtype.getVolume() * 0.5F, soundtype.getPitch() * 0.75F);
            }
        }
    }
```

### PR DESCRIPTION
**Pull request by Ecdcaeb**
Make horses and llama's fire LivingFallEvent.Fix #13 .
same as https://github.com/MinecraftForge/MinecraftForge/commit/0c2f3e55236037bd6d59cd1e18b914b7f161a63f

### CODE REVIEW

File: patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch [hunk 1]
```diff
@@ -165,23 +172,23 @@
              }
          }
      }
-@@ -343,6 +312,7 @@
+@@ -343,6 +317,7 @@
  
          this.field_110296_bG.func_110134_a(this);
          this.func_110232_cE();
 +        this.itemHandler = new net.minecraftforge.items.wrapper.InvWrapper(this.field_110296_bG);
```
Comment (Rongmario) [inline patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch:179]: Is this needed?

File: patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch [hunk 2]
```diff
@@ -137,8 +137,15 @@
 -    @Override
      public void func_180430_e(float p_180430_1_, float p_180430_2_)
      {
++        net.minecraftforge.event.entity.living.LivingFallEvent evt=net.minecraftforge.event.ForgeEventFactory.onLivingFall(this,p_180430_1_,p_180430_2_);
++        if (evt.isCanceled())return;
++        p_180430_1_=evt.getDistance();
++        p_180430_2_=evt.getDamageMultiplier();
```
Comment (Rongmario) [inline patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch:143]: Fix spacings please

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 3]
```diff
@@ -94,13 +94,7 @@
...
 import net.minecraftforge.event.entity.ThrowableImpactEvent;
 import net.minecraftforge.event.entity.item.ItemExpireEvent;
-import net.minecraftforge.event.entity.living.AnimalTameEvent;
-import net.minecraftforge.event.entity.living.LivingDestroyBlockEvent;
-import net.minecraftforge.event.entity.living.LivingEntityUseItemEvent;
-import net.minecraftforge.event.entity.living.LivingExperienceDropEvent;
-import net.minecraftforge.event.entity.living.LivingHealEvent;
-import net.minecraftforge.event.entity.living.LivingPackSizeEvent;
-import net.minecraftforge.event.entity.living.LivingSpawnEvent;
+import net.minecraftforge.event.entity.living.*;
```
Comment (Rongmario) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java:97]: Avoid star imports.

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 4]
```diff
@@ -602,6 +596,13 @@ public static float onLivingHeal(EntityLivingBase entity, float amount)
         return (MinecraftForge.EVENT_BUS.post(event) ? 0 : event.getAmount());
     }
 
+    @Nonnull
+    public static LivingFallEvent onLivingFall(EntityLivingBase entity, float distance, float damageMultiplier){
+        LivingFallEvent event=new LivingFallEvent(entity,distance,damageMultiplier);
+        MinecraftForge.EVENT_BUS.post(event);
+        return event;
+    }
+
```
Comment (Rongmario) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java:605]: Use the existing `onLivingFall` method.

File: patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch [hunk 5]
```diff
@@ -137,8 +137,15 @@
 -    @Override
      public void func_180430_e(float p_180430_1_, float p_180430_2_)
      {
++        net.minecraftforge.event.entity.living.LivingFallEvent evt=net.minecraftforge.event.ForgeEventFactory.onLivingFall(this,p_180430_1_,p_180430_2_);
++        if (evt.isCanceled())return;
```
Comment (Rongmario) [inline patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch:141]: Use the existing `onLivingFall` method.

File: patches/minecraft/net/minecraft/entity/passive/EntityLlama.java.patch [hunk 6]
```diff
@@ -0,0 +1,14 @@
+--- before/net/minecraft/entity/passive/EntityLlama.java
++++ after/net/minecraft/entity/passive/EntityLlama.java
+@@ -488,6 +488,11 @@
+     @Override
+     public void func_180430_e(float p_180430_1_, float p_180430_2_)
+     {
++        net.minecraftforge.event.entity.living.LivingFallEvent evt=net.minecraftforge.event.ForgeEventFactory.onLivingFall(this,p_180430_1_,p_180430_2_);
++        if (evt.isCanceled())return;
++        p_180430_1_=evt.getDistance();
++        p_180430_2_=evt.getDamageMultiplier();
```
Comment (Rongmario) [inline patches/minecraft/net/minecraft/entity/passive/EntityLlama.java.patch:10]: Same as above comments.

**Review by Rongmario — CHANGES_REQUESTED** [review]

File: patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch (hunk 1 above)
Comment (Ecdcaeb) [reply patches/minecraft/net/minecraft/entity/passive/AbstractHorse.java.patch:179]: idk， it is the old patch.

**Review by Rongmario — APPROVED** [review]

**Pull request merged**